src/commands/addMIConnections/SettingsAddBaseStep.ts

```diff
@@ -0,0 +1,175 @@
+            } else if (connection.name.includes('DOCUMENTDB')) {
+                await addDocumentConnectionsAndRoles(context, connection);
+            } else if (connection.name.includes('EVENTHUB')) {
+                await addEventHubConnectionsAndRoles(context, connection);
+            } else if (connection.name.includes('SERVICEBUS')) {
+                await addServiceBusConnectionsAndRoles(context, connection);
+            }
+        }
+    }
+
+    public shouldExecute(context: IAddMIConnectionsContext): boolean {
+        return !!context.connections;
+    }
+}
+
+async function getScopeHelper(context: IAddMIConnectionsContext, accountName: string, filter: string): Promise<string> {
+    const client = await createResourceClient(context);
+    const resources = await uiUtils.listAllIterator(client.resources.list({ filter: `resourceType eq '${filter}'` }));
+
+    for await (const resource of resources) {
+        if (resource.name === accountName) {
+            return nonNullProp(resource, 'id');
+        }
+    }
+
+    throw new Error(localize('noResourceFound', `No resource found with name "{0}" in subscription "{1}"`, accountName, context.subscriptionDisplayName));
+}
+
+
+
+function addRole(context: IAddMIConnectionsContext, scope: string, roleDefinition: typeof CommonRoleDefinitions[keyof typeof CommonRoleDefinitions]): void {
+    // Only assign roles if adding remote settings
+    if (context.functionapp) {
+        const role = {
+            scopeId: scope,
+            roleDefinitionId: createRoleId(context.subscriptionId, roleDefinition),
+            roleDefinitionName: roleDefinition.roleName
+        };
+
+        context.roles?.push(role);
+    }
+}
+
+async function addStorageConnectionsAndRoles(context: IAddMIConnectionsContext, connection: Connection, webJobsStorage?: boolean) {
+    // Storage connection strings are of format: DefaultEndpointsProtocol=https;AccountName=storageAccountName;AccountKey=accountKey;EndpointSuffix=core.windows.net
+    try {
+        const storageAccountName = connection.value.split(';')[1].split('=')[1];
+        context.connectionsToAdd?.push(
+            {
+                name: webJobsStorage ? 'AzureWebJobsStorage__blobServiceUri' : `${storageAccountName}__blobServiceUri`,
+                value: `https://${storageAccountName}.blob.core.windows.net`,
+            },
+            {
+                name: webJobsStorage ? 'AzureWebJobsStorage__queueServiceUri' : `${storageAccountName}__queueServiceUri`,
+                value: `https://${storageAccountName}.queue.core.windows.net`,
+            },
+            {
+                name: webJobsStorage ? 'AzureWebJobsStorage__tableServiceUri' : `${storageAccountName}__tableServiceUri`,
+                value: `https://${storageAccountName}.table.core.windows.net`,
+            },
+            ...getClientIdAndCredentialProperties(context, webJobsStorage ? 'AzureWebJobsStorage' : storageAccountName)
```

Comment:
I think you can include this in the `if (context.functionApp)` too. I believe it's only called in a remote situation as well.